src/main/webapp/app/exam/shared/working-time-update/working-time-control.component.ts

```diff
@@ -0,0 +1,165 @@
+        this.currentExam = exam;
+        this.initWorkingTimeFromCurrentExam();
+    }
+
+    get exam(): Exam | undefined {
+        return this.currentExam;
+    }
+
+    private currentExam?: Exam;
+    private touched = false;
+    private onTouched = () => {};
+    private onChange: (_: number) => void = () => {};
+
+    workingTime = {
+        hours: 0,
+        minutes: 0,
+        seconds: 0,
+        percent: 0,
+    };
+
+    constructor(private artemisDurationFromSecondsPipe: ArtemisDurationFromSecondsPipe) {}
+
+    // ControlValueAccessor implementation
+
+    /**
+     * Updates the working time duration inputs whenever
+     * the value of the form control changes.
+     * @param seconds
+     */
+    writeValue(seconds: number) {
+        if (seconds) {
+            this.setWorkingTimeDuration(seconds);
+            this.updateWorkingTimePercentFromDuration();
+        }
+    }
+
+    registerOnChange(onChange: any) {
+        this.onChange = onChange;
+    }
+
+    registerOnTouched(onTouched: any) {
+        this.onTouched = onTouched;
+    }
+
+    setDisabledState(disabled: boolean) {
+        this.disabled = disabled;
+    }
+
+    private markAsTouched() {
+        if (!this.touched) {
+            this.onTouched();
+            this.touched = true;
+        }
+    }
+
+    /**
+     * Updates the controls based on the working time of the student exam.
+     */
+    private initWorkingTimeFromCurrentExam() {
+        if (this.exam) {
+            this.setWorkingTimeDuration(normalWorkingTime(this.exam)!);
+            this.updateWorkingTimePercentFromDuration();
+            this.onChange(this.getWorkingTimeSeconds());
+        }
+    }
+
+    /**
+     * Updates the working time duration values of the control whenever the percent value was changed.
+     * After the update, the onChange callback is called with the new working time in seconds.
+     */
+    onPercentChanged() {
+        this.markAsTouched();
+        this.updateWorkingTimeDurationFromPercent();
+        this.onChange(this.getWorkingTimeSeconds());
+    }
+
+    /**
+     * Updates the working time percent value of the control whenever the duration values were changed.
+     * After the update, the onChange callback is called with the new working time in seconds.
+     */
+    onDurationChanged() {
+        this.markAsTouched();
+        this.updateWorkingTimePercentFromDuration();
+        this.onChange(this.getWorkingTimeSeconds());
+    }
+
+    /**
+     * Updates the working time percent value of the control based on the current working time duration.
+     * @private
+     */
+    private updateWorkingTimePercentFromDuration() {
+        if (this.exam) {
+            this.workingTime.percent = getRelativeWorkingTimeExtension(this.exam, this.getWorkingTimeSeconds());
+        }
+    }
+
+    /**
+     * Updates the working time duration values of the control based on the current working time percent.
+     * @private
+     */
```

Comment:
```suggestion
    /**
     * Updates the working time percent value of the control based on the current working time duration.
     */
    private updateWorkingTimePercentFromDuration() {
        if (this.exam) {
            this.workingTime.percent = getRelativeWorkingTimeExtension(this.exam, this.getWorkingTimeSeconds());
        }
    }

    /**
     * Updates the working time duration values of the control based on the current working time percent.
     */
```
`@private` is a JSDoc feature that is meant to compensate for missing typing in plain JavaScript. It has no effect in TypeScript and is redudant since the method is already private. This also might lead to inconsistencies if the method gets public in the future, but the documentation does not get changed.